pkg/sql/backfill/backfill.go

```diff
@@ -588,6 +614,7 @@ func (ib *IndexBackfiller) BuildIndexEntriesChunk(
 	// indexes so use a smaller value.
 	const initBufferSize = 1000
 	entries := make([]rowenc.IndexEntry, 0, initBufferSize*int64(len(ib.added)))
+	ib.boundAccount.Clear(ctx)
```

Comment:
Hmm, this seems wrong to me - you've just made a large buffer, I'd expect you to be accounting for the memory of that buffer.